File: Desafio_Rivia/3_read_from_api_write_to_json.py
import requests
import json

def consumir_api_e_escrever_json(api_url, json_arquivo):
    response = requests.get(api_url)
    dados = response.json()

    with open(json_arquivo, mode='w') as arquivo_json:
        json.dump(dados, arquivo_json, indent=4)

# Exemplo de uso (URL fictícia)
api_url = 'https://api.correios.com.br/rastreio/codigo'
consumir_api_e_escrever_json(api_url, 'dados_api.json')

File: Desafio_Rivia/3_read_from_api_write_to_json.js
const axios = require('axios');
const fs = require('fs');

// Função para consumir API e escrever o resultado em um arquivo JSON
async function consumirApiEscreverJson(apiUrl, jsonArquivo) {
    try {
        const response = await axios.get(apiUrl); // Faz a requisição GET na API
        const dados = response.data;              // Extrai os dados da resposta

        // Escreve os dados no arquivo JSON
        fs.writeFileSync(jsonArquivo, JSON.stringify(dados, null, 4), 'utf8');
        console.log(`Dados salvos no arquivo: ${jsonArquivo}`);
    } catch (error) {
        console.error(`Erro ao consumir a API: ${error}`);
    }
}

// Exemplo de uso (URL fictícia)
const apiUrl = 'https://api.correios.com.br/rastreio/codigo';
consumirApiEscreverJson(apiUrl, 'dados_api.json');
